packages/wdio-config/src/utils.js

```diff
@@ -47,6 +47,33 @@ export function isCucumberFeatureWithLineNumber(spec) {
     return specs.some((s) => s.match(/:\d+(:\d+$|$)/))
 }
 
+function isBrowserstackBackend(options) {
+    let { user, key } = options
+    return typeof user === 'string' && typeof key === 'string' && key.length === 20
+}
+
+function isTestingBotBackend(options) {
+    let { user, key } = options
+    return typeof user === 'string' && typeof key === 'string' && key.length === 32
+}
+
+function isSauceLabsBackend(options, isRDC) {
+    let { user, key } = options
+    return (typeof user === 'string' && typeof key === 'string' && key.length === 36) ||
+    // When SC is used a user needs to be provided and `isRDC` needs to be true
+    (typeof user === 'string' && isRDC) ||
+    // Or only RDC
+    isRDC
+}
+
+export function isCloudCapability(cap) {
+    return cap && (cap['bstack:options'] || cap['sauce:options'] || cap['tb:options'])
+}
+
+export function isCloudBackend(options, isRDC) {
+    return isBrowserstackBackend(options) || isTestingBotBackend(options) || isSauceLabsBackend(options, isRDC)
```

Comment:
This function is not used at the moment. It's added for reference to my previous approach, instead of detecting capability, we detect backend. I can remove if you think this is not necessary.